test/integration/tlsroute_test.go

```diff
@@ -707,29 +719,25 @@ func TestTLSRoutePassthrough(t *testing.T) {
 // an explanation if it is not (typical network related errors like io.EOF or
 // syscall.ECONNRESET are returned directly).
 func tlsEchoResponds(
-	url string, podName string, hostname, certHostname string, passthrough bool,
+	url string, podName string, hostname string, certPool *x509.CertPool, passthrough bool,
 ) error {
 	dialer := net.Dialer{Timeout: time.Second * 10}
 	conn, err := tls.DialWithDialer(&dialer,
 		"tcp",
 		url,
 		&tls.Config{
-			ServerName:         hostname,
-			InsecureSkipVerify: true,
+			MinVersion: tls.VersionTLS12,
+			ServerName: hostname,
+			RootCAs:    certPool,
 		})
 	if err != nil {
 		return err
 	}
 	defer conn.Close()
 
-	cert := conn.ConnectionState().PeerCertificates[0]
-	if cert.Subject.CommonName != certHostname {
-		return fmt.Errorf("expected certificate with cn=%s, got cn=%s", certHostname, cert.Subject.CommonName)
-	}
-
 	header := []byte(fmt.Sprintf("Running on Pod %s.", podName))
-	// if we are testing with passthrough, the go-echo service should return a message
-	// noting that it is listening in TLS mode.
+	// If we are testing with passthrough, the go-echo service should return a message
+	// nothing that it is listening in TLS mode.
```

Comment:
This is incorrect I believe.

We can make this more explicit via e.g. 

```suggestion
	// mentioning that it is listening in TLS mode.
```